paimon-core/src/main/java/org/apache/paimon/operation/AbstractFileStoreWrite.java

```diff
@@ -369,6 +373,14 @@ public CompactionMetrics getCompactionMetrics(BinaryRow partition, int bucket) {
         return null;
     }
 
+    @Nullable
+    public WriterMetrics getWriterMetrics() {
+        if (metricRegistry != null && writerMetrics == null) {
+            writerMetrics = new WriterMetrics(metricRegistry, tableName);
+        }
+        return writerMetrics;
+    }
```

Comment:
No need for lazy allocation. Just create a new `writeMetrics` in `withMetricRegistry`.